BookStore.h

```diff
@@ -0,0 +1,135 @@
+#ifndef UNTITLED_BOOKSTORE_H
+#define UNTITLED_BOOKSTORE_H
+#ifndef UNTITLED_H
+#define UNTITLED_H
+
+#include "string"
+#include "vector"
+#include "algorithm"
+#include "iostream"
+
+class Book {
+public:
+    Book(std::string name,
+         int price_,
+         int numberOfPages_,
+         std::string author_,
+         int quantity_,
+         int numberOfSales_);
+    ~Book();
+
+    std::string getName(){
+        return name;
+    }
+    int getPrice(){
+        return price;
+    }
+    int getNumberOfSales(){
+        return numberOfSales;
+    }
+    std::string getAuthor(){
+        return author;
+    }
+
+private:
+    std::string name;
+    int price;
+    int numberOfPages;
+    std::string author;
+    int quantity;
+    int numberOfSales;
+};
+
+class BookStore {
+public:
+    BookStore(std::vector<Book> books);
+    BookStore();
+    ~BookStore();
+
+    void addBook(Book book){
+        books.push_back(book);
+    }
+
+    void deleteBook(std::string name) {
+        int ind = 0;
+        for (int i = 0; i < books.size(); ++i) {
+            if (books[i].getName() == name) {
+                ind = i;
+                break;
+            }
+        }
+        for (int i = ind; i < books.size()-1; ++i) {
+            books[i] = books[i + 1];
+        }
+        books.pop_back();
+    }
+
+    void sortBooksByPrice(){
+        if (books.size()==0){
+            std::cout << "The store is empty!\n";
+            return;
+        }
+        auto comp = [&] (Book a, Book b){
+            return a.getPrice() > b.getPrice();
+        };
+        std::vector<Book> arr = books;
+        std::sort(arr.begin(), arr.end(), comp);
+        std::cout << "--------------------------\n";
+        for (int i = 0; i < arr.size(); ++i) {
+            std::cout << i+1 << ". " <<
+            arr[i].getName() << " | " <<
+            arr[i].getAuthor() << " | " <<
+            arr[i].getPrice() << " | " <<
+            arr[i].getNumberOfSales() << "\n";
+        }
+        std::cout << "--------------------------\n";
+    }
+
+    void sortBooksByNumberOfSales(){
```

Comment:
instead of sortBooksByNumberOfSales, the function could be named sortBooksByNumberOfSalesAscending or sortBooksByNumberOfSalesDescending as it is more descriptive.

Reply:
changed the name